king_phisher/server/fs_utilities.py

```diff
@@ -56,6 +58,72 @@ def chown(path, user=None, group=constants.AUTOMATIC, recursive=True):
 	:type group: int, str, ``None``, :py:class:`~king_phisher.constants.AUTOMATIC`
 	:param bool recursive: Whether or not to recurse into directories.
 	"""
+
+	user, group = _resolve_information(user, group)
+
+	if recursive:
+		iterator = smoke_zephyr.utilities.FileWalker(path)
+	else:
+		iterator = (path,)
+	for path in iterator:
+		os.chown(path, user, group)
+
+
+def access(path, mode, user=None, group=constants.AUTOMATIC):
+	"""
+	This is a high-level wrapper around :py:func:`os.access` to provide
+	additional functionality. Similar to `os.access` this high-level wrapper
+	will test the given path for a variety of access modes. Additionally however,
+	*user* or *group* can be specified to test against a specific of user or group.
+
+	.. versionadded:: 1.14.0
+
+	:param str path: The path to test access for.
+	:param str mode: The mode to test access for. Set to `R_OK` to test for readability,
+		`W_OK` for writability and
+		`X_OK` to determine if path can be executed.
+	:param user: The user to test permissions for.
+	:type user: int, str, ``None``
+	:param group: The group to test permissions for. If set to
+		:py:class:`~king_phisher.constants.AUTOMATIC`, the group that *user*
+		belongs too will be used.
+	:type group: int, str, ``None``, :py:class:`~king_phisher.constants.AUTOMATIC`
+	:return: Returns ``True`` only if the user or group has the mode of permission specified else returns ``False``.
+	:rtype: bool
+	"""
+
+	user_id, group = _resolve_information(user, group)
+	file_info = os.stat(path)
+
+	if mode == 'R_OK':
```

Comment:
If you're going to define, user, group, and other permissions this way, then the values should be initialized to 0 and then you should set the bits based on the mode flags like:
```python
user_permissions = 0
group_permissions = 0
other_permissions = 0
if mode & os.R_OK:
    user_permissions |= stat.S_IRUSR
    group_premissions |= stat.S_IRGRP
    other_permissions |= stat.S_IROTH
if mode & os.W_OK:  # note that this is `if` not `elsif` so flags can be `os.R_OK | os.W_OK`
    user_permissions |= stat.S_IWUSR
    # and so on and so forth
```

Then later when you go to check them, you'd need to make sure all the bits are equal doing:
```python
(file_info.st_mode & user_permissions == user_permissions)
```
This will ensure that all the flags set in `user_permissions` are present and not just one of them (which would be enough to cause the result to be non-zero and return true. This will still have the short coming of requiring that the permissions are all present for either the user, group or other but I think that's fine so long as we document it.